### PR TITLE
Upgrade ng 12

### PR DESCRIPTION
**Pull request by dinaNW**
https://newswhip.atlassian.net/browse/RED-880
https://testspike3.gcp.newswhip.com/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,29 +25,25 @@
...
     "@fortawesome/fontawesome-pro": "^5.7.2",
     "d3": "^5.7.0",
     "ngx-bootstrap": "^6.2.0"
   },
   "devDependencies": {
-    "@angular-devkit/build-angular": "~0.1102.3",
-    "@angular/animations": "^11.2.4",
+    "@angular-devkit/build-angular": "~12.1.3",
+    "@angular/animations": "^12.1.3",
     "@angular/cdk": "^11.2.4",
```
Comment (garethdn) [inline package.json:36]: This should probably go to v12

File: package.json [hunk 2]
```diff
@@ -25,29 +25,25 @@
...
   "private": true,
   "peerDependencies": {
-    "@angular/animations": "^11.2.4",
     "@angular/cdk": "^11.2.4",
-    "@angular/common": "^11.2.4",
-    "@angular/core": "^11.2.4",
-    "@angular/forms": "^11.2.4",
     "@fortawesome/fontawesome-pro": "^5.7.2",
     "d3": "^5.7.0",
     "ngx-bootstrap": "^6.2.0"
```
Comment (garethdn) [inline package.json:31]: According to https://github.com/valor-software/ngx-bootstrap#compatibility we can and probably should move to v7 - but you should see what changes are in v7 to see if there's anything we need to update. This comment applies to Spike also. If you do update you'll need to update this reference in peerDependencies and the other in devDependencies

**Review by garethdn — CHANGES_REQUESTED** [review]
CDK should definitely be upgraded - ngx-bootstrap probably should but it could be dependent on effort

File: src/_lib/sass/src/_feature-alerts.scss [hunk 3]
```diff
@@ -29,19 +29,19 @@
 
     &,
     &.hotspot-left {
-        left: -#{$hotspot-size / 2};
-        top: calc(50% - #{$hotspot-size / 2});
+        left: -#{$hotspot-size * 0.5};
```
Comment (dinaNW) [inline src/_lib/sass/src/_feature-alerts.scss:32]: Used https://sass-lang.com/documentation/cli/migrator for these.

File: src/_lib/sass/styles.scss [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
 // Utilities
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/hide-text";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/opacity";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/image";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/responsive-visibility";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/reset-text";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/text-emphasis";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/text-overflow";
-@import "node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/vendor-prefixes";
+@import "../../../node_modules/bootstrap-sass/assets/stylesheets/bootstrap/mixins/hide-text";
```
Comment (dinaNW) [inline src/_lib/sass/styles.scss:4]: Not sure what caused these paths to break, but simply saying `node_modules/...` no longer works.

**Review by garethdn — APPROVED** [review]

**Pull request merged**